### PR TITLE
gawk: skip check if cmp unavailable

### LINKED ISSUE #318: gawk depends on diffutils
According to https://gist.github.com/minektur/29694ec4b41e33e7c79b520d0145a484#file-output-L249 installing `gawk` from source requires working `cmp` provided by `diffutils` for `make check` to complete successfully.

### PR DESCRIPTION
**Pull request by DoomHammer**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----

Fixes #318

### CODE REVIEW

File: Formula/gawk.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 class Gawk < Formula
   desc "GNU awk utility"
   homepage "https://www.gnu.org/software/gawk/"
-  url "http://ftpmirror.gnu.org/gawk/gawk-4.1.3.tar.xz"
-  mirror "https://ftp.gnu.org/gnu/gawk/gawk-4.1.3.tar.xz"
+  url "https://ftpmirror.gnu.org/gawk/gawk-4.1.3.tar.xz"
```
Comment (sjackman) [inline Formula/gawk.rb]: Why did you delete the mirror? Did `brew audit` suggest this change?

Comment (DoomHammer) [inline Formula/gawk.rb]: Yes, it did.

File: Formula/gawk.rb [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
   sha256 "e3cf55e91e31ea2845f8338bedd91e40671fc30e4d82ea147d220e687abda625"
 
   bottle do
+    revision 2
```
Comment (sjackman) [inline Formula/gawk.rb]: Revert this line before committing. This PR looks good otherwise.

**Comment by sjackman** [comment]
Please rebase this PR onto `master` delete the revision, go ahead and merge! Thanks, Piotr.

**Comment by DoomHammer** [comment]
That's the plan :)

**Comment by sjackman** [comment]
Feel free to `linuxbrew build-bottle-pr` anything on that list that suits your fancy as well.
